dotnet/samples/Demos/BedrockTest/Program.cs

```diff
@@ -0,0 +1,225 @@
+    var streamTextGeneration = streamTextGenerationService.GetStreamingTextContentsAsync(userStreamTextPrompt).ConfigureAwait(true);
+    await foreach (var textContent in streamTextGeneration)
+    {
+        Console.Write(textContent.Text);
+    }
+
+    Console.WriteLine();
+}
+
+// List of available models
+Dictionary<int, string> GetBedrockModels()
+{
+    return new()
+    {
+        { 1, "anthropic.claude-v2" },
+        { 2, "anthropic.claude-v2:1" },
+        { 3, "anthropic.claude-instant-v1" },
+        { 4, "anthropic.claude-3-sonnet-20240229-v1:0" },
+        { 5, "anthropic.claude-3-haiku-20240307-v1:0" },
+        { 6, "cohere.command-light-text-v14" },
+        { 7, "cohere.command-text-v14" },
+        { 8, "cohere.command-r-v1:0" },
+        { 9, "cohere.command-r-plus-v1:0" },
+        { 10, "ai21.jamba-instruct-v1:0" },
+        { 11, "ai21.j2-mid-v1" },
+        { 12, "ai21.j2-ultra-v1" },
+        { 13, "meta.llama3-8b-instruct-v1:0" },
+        { 14, "meta.llama3-70b-instruct-v1:0" },
+        { 15, "mistral.mistral-7b-instruct-v0:2" },
+        { 16, "mistral.mixtral-8x7b-instruct-v0:1" },
+        { 17, "mistral.mistral-large-2402-v1:0" },
+        { 18, "mistral.mistral-small-2402-v1:0" },
+        { 19, "amazon.titan-text-lite-v1" },
+        { 20, "amazon.titan-text-express-v1" },
+        { 21, "amazon.titan-text-premier-v1:0" }
+    };
+}
+
+// Get the user's model choice
+int GetUserChoice()
+{
+    int pick;
+
+    // Display the available options
+    Console.WriteLine("Choose an option:");
+    Console.WriteLine("1. Chat Completion");
+    Console.WriteLine("2. Text Generation");
+    Console.WriteLine("3. Stream Chat Completion");
+    Console.WriteLine("4. Stream Text Generation");
+
+    Console.Write("Enter your choice (1-4): ");
+    while (!int.TryParse(Console.ReadLine(), out pick) || pick < 1 || pick > 4)
+    {
+        Console.WriteLine("Invalid input. Please enter a valid number from the list.");
+        Console.Write("Enter your choice (1-4): ");
+    }
+
+    return pick;
+}
+
+int GetModelNumber(List<int> modelsToSkip, string serviceType)
+{
+    int chosenModel;
+
+    // Display the model options
+    Console.WriteLine($"Available {serviceType} models:");
+    foreach (var option in bedrockModels)
+    {
+        Console.WriteLine($"{option.Key}. {option.Value}");
+    }
+
+    Console.Write($"Enter the number of the model you want to use for {serviceType}: ");
+    while (!int.TryParse(Console.ReadLine(), out chosenModel) || !bedrockModels.ContainsKey(chosenModel) || modelsToSkip.Contains(chosenModel))
+    {
+        Console.WriteLine($"Invalid input. Please enter a valid number from the list ({(modelsToSkip.Count > 0 ? $"Models {string.Join(", ", modelsToSkip)} are not supported" : string.Empty)}).");
+        Console.Write($"Enter the number of the model you want to use for {serviceType}: ");
+    }
+
+    return chosenModel;
+}
```

Comment:
Skipping using a list is hard to maintain as new amazon models can be made available. Suggest adding a struct and use the dictionary with that information.

While getting the list of available models in each function use a `Linq.Where()` to get what you need.

```suggestion
}

private struct ModelDefinition {
    internal SupportedModality Modality { get; set; }
    internal string Name { get; set; }
    internal bool CanStream { get; set; }
    
    
    internal enum SupportedModality {
        TextCompletion,
        ChatCompletion
    }
}
```